### PR TITLE
UploadDialog: allow removing assets before uploading

### PR DESCRIPTION
**Pull request by gu-stav**
### What does it do?

It allows removing assets, after there were selected for uploading but before actually uploading them.

Screenshot from 2022-02-07 22-09-57

### Why is it needed?

It allows users to review the selected assets before uploading them.

### How to test it?

1. Go to the media library
2. Add new assets
3. Select files from your computer
4. You should now be able to edit/ remove assets

### Related issue(s)/PR(s)

- https://strapi-inc.atlassian.net/browse/CONTENT-62

### CODE REVIEW

File: packages/core/upload/admin/src/components/AssetCard/ImageAssetCard.js [hunk 1]
```diff
@@ -16,13 +16,22 @@ import {
...
 import Pencil from '@strapi/icons/Pencil';
+import Trash from '@strapi/icons/Trash';
 import { useIntl } from 'react-intl';
 import { getTrad } from '../../utils';
 
 const Extension = styled.span`
   text-transform: uppercase;
 `;
 
+const CardActions = styled(CardAction)`
```
Comment (ronronscelestes) [inline packages/core/upload/admin/src/components/AssetCard/ImageAssetCard.js:27]: if having multiple items in CardActions is a real need / documented in the DS, maybe we could think about integrating the design related to this need on the DS directly, instead of having to customize the component, WDYT? cc @HichamELBSI @maevalienard

**Review by ronronscelestes — COMMENTED** [review]
Looks good!
Wrote a small question about having multiple actions on the top right on the Card / inside CardActions component

File: packages/core/upload/admin/src/components/AssetCard/ImageAssetCard.js (hunk 1 above)
Comment (maevalienard) [reply packages/core/upload/admin/src/components/AssetCard/ImageAssetCard.js:27]: I don't have any opinion to be honest :D

Comment (gu-stav) [reply packages/core/upload/admin/src/components/AssetCard/ImageAssetCard.js:27]: @HichamELBSI That's nice 🌻 . Thanks! Didn't know this was possible with `styled-components`.

Comment (gu-stav) [reply packages/core/upload/admin/src/components/AssetCard/ImageAssetCard.js:27]: PR is there: https://github.com/strapi/design-system/pull/508 Let's move the discussion over there. Thanks everyone!

Comment (HichamELBSI) [reply packages/core/upload/admin/src/components/AssetCard/ImageAssetCard.js:27]: Many thanks!!

**Review by HichamELBSI — APPROVED** [review]
Great job!

**Comment by gu-stav** [comment]
This PR is currently blocked until https://github.com/strapi/strapi/pull/12473 is merged.

**Comment by gu-stav** [comment]
Rebased the branch on master, so that it should contain the design-system update now.

**Review by HichamELBSI — DISMISSED** [review]
LGTM

File: packages/core/upload/admin/src/components/AssetCard/tests/ImageAssetCard.test.js.orig [hunk 2]
```diff
@@ -0,0 +1,702 @@
+import React from 'react';
```
Comment (HichamELBSI) [inline packages/core/upload/admin/src/components/AssetCard/tests/ImageAssetCard.test.js.orig:1]: Is this file needed?

Comment (gu-stav) [reply packages/core/upload/admin/src/components/AssetCard/tests/ImageAssetCard.test.js.orig]: No. You are right. It's a leftover from a bad merge. Good catch, thank you!

Removed in 48e60db5c36b7bc0225a39363928021ab25bccee

**Review by HichamELBSI — APPROVED** [review]
LGTM
